framework/tlvf/tlvf.py

```diff
@@ -1009,8 +1009,8 @@ def addClassParam(self, obj_meta, param_name, param_type, param_type_info, param
 
                 lines_h.append( "bool set_%s(const char buffer[], size_t size);" % (param_name) )
                 lines_cpp.append( "bool %s::set_%s(const char str[], size_t size) {" % (obj_meta.name, param_name) )
-                lines_cpp.append( "%sif (str == nullptr || size == 0) {" % self.getIndentation(1))
-                lines_cpp.append( '%sTLVF_LOG(WARNING) << "set_%s received an empty string.";' %  (self.getIndentation(2), param_name) )
+                lines_cpp.append( "%sif (str == nullptr) {" % self.getIndentation(1))
+                lines_cpp.append( '%sTLVF_LOG(WARNING) << "set_%s received a null pointer.";' %  (self.getIndentation(2), param_name) )
```

Comment:
See the comment below in this file. Let me know if you still think it's better to just add a check above line 1022 and return true prematurely. I'll do it in a separate pull request then since this change will trigger tons of changes in the auto-generated committed files which probably will make this pull request harder to deal with.